fish_prompt.fish

```diff
@@ -801,6 +802,20 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
     set_color normal
 end
 
+function __bobthefish_prompt_nix -S -d 'Display current nix environment'
+    [ "$theme_display_nix" = 'no' -o -z "$IN_NIX_SHELL" ]
+    and return
+
+    if [ "$IN_NIX_SHELL" = "impure" ]
+        set say_what "nix"
+    else
+        set say_what "pure"
+    end
+
+    __bobthefish_start_segment $color_nix
+    echo -ns $nix_glyph' ' "$say_what" ' '
```

Comment:
normally we put the trailing space on the glyph itself, when the glyph is optional (e.g. when it's from geek fonts). otherwise we end up with awkward spacing when the glyph is removed.